src/app/DTT/upload/dropzone.js

```diff
@@ -1,45 +1,63 @@
 import { useState } from 'react'
 import { useDropzone } from 'react-dropzone'
 import { Box, Flex, Text, Input, Button, Center, Image, useToast } from '@chakra-ui/react'
-import { isEmpty, map } from 'ramda'
+import { isEmpty, map, pathOr, compose, includes, split } from 'ramda'
 
-const DropZone = ({ handleSave, closeDropzone, maxSize = '5000000' }) => {
+const DropZone = ({ handleSave, closeDropzone, maxSize = '5000000', maxFiles = 1 }) => {
   const [files, setFiles] = useState([])
   const toast = useToast()
+  const checkIfImage = compose(includes('image'), split('/'))
+
+  const showErrorMessage = file => {
+    const errorMessage = pathOr('', ['errors', 0, 'message'])(file)
+    return toast({
+      position: 'bottom',
+      title: 'An error occurred.',
+      description: errorMessage,
+      status: 'error',
+      duration: 3000,
+    })
+  }
 
   const { getRootProps, getInputProps } = useDropzone({
-    accept: 'image/*',
+    accept: 'image/*, application/pdf',
     maxSize: maxSize,
+    maxFiles: maxFiles,
     onDrop: (acceptedFiles, rejectedFiles) => {
       setFiles(
         acceptedFiles.map(file =>
           Object.assign(file, {
             preview: URL.createObjectURL(file),
           }),
         ),
-        rejectedFiles.map(file =>
-          toast({
-            position: 'bottom',
-            title: 'An error occurred.',
-            description: `There was an error, try again!`,
-            status: 'error',
-            duration: 3000,
-          }),
-        ),
+        !isEmpty(rejectedFiles) && showErrorMessage(rejectedFiles[0]),
       )
     },
   })
 
-  const preview = map(file => (
-    <Box w="200px" h="200px" key={file.name}>
-      <Image src={file.preview} alt="drag and drop" boxSize="200px" objectFit="cover" />
-    </Box>
-  ))(files)
+  const preview = map(({ name, preview, type }) => {
+    if (checkIfImage(type)) {
+      return (
+        <Box w="200px" h="200px" key={name} r="5">
+          <Image src={preview} alt="drag and drop" boxSize="200px" objectFit="cover" />
+        </Box>
+      )
+    }
+
+    return (
+      <Flex key={name} p="4" background="teal.50">
+        <Text color="black">{`Uploaded File:`}</Text>
```

Comment:
black text is hardcoded? what if we are in dark mode will these colors all work?

Reply:
The background is hardcoded to be white for the component irrespective of the mode. 😅